src/ios/GeofencePlugin.swift

```diff
@@ -324,13 +324,16 @@ class GeoNotificationManager : NSObject, CLLocationManagerDelegate {
         log("Monitoring region " + region.identifier + " failed " + error.description)
     }
 
-    func handleTransition(region: CLRegion!) {
+    func handleTransition(region: CLRegion!, type: Int) {
         if let geo = store.findById(region.identifier) {
             if let notification = geo["notification"].asDictionary {
                 notifyAbout(geo)
             }
 
-            NSNotificationCenter.defaultCenter().postNotificationName("handleTransition", object: geo.description)
+            var d = geo.description
+            d = d.substringToIndex(advance(d.startIndex, count(d) - 1)) + ",\"handledTransition\":\(type)}"
```

Comment:
I think it would be better to just change fetched geofence object transitionType like `geo["transitionType"] = type`